services/src/datasets/external/netcdfcf/mod.rs

```diff
@@ -1561,12 +1737,134 @@ impl MetaDataProvider<OgrSourceDataset, VectorResultDescriptor, VectorQueryRecta
     }
 }
 
+fn netcfg_gdal_path(
+    base_path: Option<&Path>,
+    path: &Path,
+    group_path: &[String],
+) -> Result<PathBuf> {
+    let path = if let Some(base_path) = base_path {
+        canonicalize_subpath(base_path, path).map_err(|_| {
+            NetCdfCf4DProviderError::FileIsNotInProviderPath {
+                file: path.to_string_lossy().into(),
+            }
+        })?
+    } else {
+        path.to_owned()
+    };
+
+    let path = path.to_string_lossy().to_string();
+    let group_path = group_path.join("/");
+    let gdal_path = format!("NETCDF:{path}:/{group_path}/ebv_cube");
+    Ok(PathBuf::from(gdal_path))
+}
+
+fn gdal_netcdf_open(base_path: Option<&Path>, path: &Path) -> Result<gdal::Dataset> {
+    let path = if let Some(base_path) = base_path {
+        canonicalize_subpath(base_path, path).map_err(|_| {
+            NetCdfCf4DProviderError::FileIsNotInProviderPath {
+                file: path.to_string_lossy().into(),
+            }
+        })?
+    } else {
+        path.to_owned()
+    };
+
+    let dataset = gdal_open_dataset_ex(
+        &path,
+        DatasetOptions {
+            open_flags: GdalOpenFlags::GDAL_OF_READONLY | GdalOpenFlags::GDAL_OF_MULTIDIM_RASTER,
+            allowed_drivers: Some(&["netCDF"]),
+            open_options: None,
+            sibling_files: None,
+        },
+    )
+    .context(error::InvalidDatasetIdFile)?;
+
+    Ok(dataset)
+}
+
+#[cfg(test)]
+pub(crate) fn test_db_config() -> DatabaseConnectionConfig {
+    let db_config =
+        crate::util::config::get_config_element::<crate::util::config::Postgres>().unwrap();
+    DatabaseConnectionConfig {
+        host: db_config.host,
+        port: db_config.port,
+        database: db_config.database,
+        schema: db_config.schema,
+        user: db_config.user,
+        password: db_config.password,
+    }
+}
+
+#[cfg(test)]
+async fn test_db() -> Pool<PostgresConnectionManager<NoTls>> {
+    let db = Pool::builder()
+        .max_size(1) // unwised to have to separate connections point to `pg_temp`
+        .build(PostgresConnectionManager::new(
+            test_db_config().pg_config(),
+            NoTls,
+        ))
+        .await
+        .unwrap();
+
+    migrate_database(&mut db.get().await.unwrap(), &all_migrations())
+        .await
+        .unwrap();
+
+    db
+}
+
+#[cfg(test)]
+async fn test_db_connection(
+) -> bb8_postgres::bb8::PooledConnection<'static, PostgresConnectionManager<NoTls>> {
+    test_db().await.get_owned().await.unwrap()
+}
+
+/// Creates a write transaction that defers constraints
+async fn deferred_write_transaction<'db, 't>(
```

Comment:
just for curiousity: why do you need to defer the stuff? Deferring is not free (performance-wise) as far as I understand it.

Reply:
It is super expensive to check all the constraints during each insert.
What we do here is more of a bulk insert style with different statements.
Either I drop all the constraints or I just check them once at the end.